batcher/aligned-batcher/src/lib.rs

```
@@ -635,13 +637,22 @@ impl Batcher {
         }
 
         if self.is_nonpaying(&addr) {
-            // TODO: Non paying msg and paying should share some logic
-            return self
-                .handle_nonpaying_msg(ws_conn_sink.clone(), &client_msg)
+            info!("Generating non-paying data");
+            let Ok(non_paying_data) = self.generate_non_paying_data(&client_msg).await else {
+                error!("Failed to generate non paying data");
+                send_message(
+                    ws_conn_sink.clone(),
+                    SubmitProofResponseMessage::NonPayingAddressError,
+                )
                 .await;
+                return Ok(());
+            };
```

This can only fail if the non-paying address is not allowed, shouldn't we return that error instead of a generic one? Or maybe `NonPayingAddressError` can take the error in the enum (i.e NonPayingAddressError(TransactionError)).